File: backends/fastapi/app/schemas/capsule_recommendation.py
"""胶囊 AI 推荐主题 DTO。"""

from __future__ import annotations

from pydantic import BaseModel, Field


class CapsuleRecommendationItem(BaseModel):
    title: str = Field(min_length=1, max_length=60)
    hint: str = Field(min_length=1, max_length=80)
    openInDays: int = Field(ge=1, le=3650)  # noqa: N815


class CapsuleRecommendationList(BaseModel):
    items: list[CapsuleRecommendationItem]
    generatedBy: str  # noqa: N815
    cached: bool = False

File: backends/gin/internal/dto/capsule_recommendation.go
package dto

// CapsuleRecommendationItem 单条 AI 推荐主题（仅元数据，不含完整胶囊）。
type CapsuleRecommendationItem struct {
	Title      string `json:"title"`
	Hint       string `json:"hint"`
	OpenInDays int    `json:"openInDays"`
}

// CapsuleRecommendationList 推荐列表响应体。
// 推荐为锦上添花：LLM 不可用时 Items 为空数组（非 null），GeneratedBy 为 "none"。
type CapsuleRecommendationList struct {
	Items       []CapsuleRecommendationItem `json:"items"`
	GeneratedBy string                      `json:"generatedBy"`
	Cached      bool                        `json:"cached"`
}
